### PR TITLE
Tools: Move fpgainfo boot page info to board specific modules

### PR DESCRIPTION
**Pull request by anandaravuri**
-boot page info is different  from N3000 card and N6010 cards,

 -N3000 boot page info  Factory or user

  N61010 boot page Factory or user1 or user2
  Read power_on_image  syfs string from DFL MAX10 feature and find boot order from sysfs string.

  -cat /sys/class/fpga_region/region0/dfl-fme.0/dfl_dev.2/n6010bmc-secure.2.auto/power_on_image  fpga_user2 fpga_user1 fpga_factory

Signed-off-by: anandaravuri <ananda.ravuri@intel.com>

### CODE REVIEW

File: tools/libboard/board_n6010/board_n6010.c [hunk 1]
```diff
@@ -519,4 +525,43 @@ fpga_result print_fme_verbose_info(fpga_token token)
...
+		reg_res = regcomp(&re, BOOTPAGE_PATTERN, REG_EXTENDED | REG_ICASE);
+		if (reg_res) {
+			OPAE_ERR("Error compiling regex");
+			return FPGA_EXCEPTION;
+		}
+
+		reg_res = regexec(&re, boot, 3, matches, 0);
+		if (reg_res) {
+			regerror(reg_res, &re, err, sizeof(err));
+			OPAE_MSG("Error executing regex: %s", err);
```
Comment (tswhison) [inline tools/libboard/board_n6010/board_n6010.c:555]: regfree(&re) ?

File: tools/libboard/board_n6010/board_n6010.c [hunk 2]
```diff
@@ -519,4 +525,43 @@ fpga_result print_fme_verbose_info(fpga_token token)
...
+			return FPGA_EXCEPTION;
+		}
+		strncpy(page, boot + matches[0].rm_so+1,
+			matches[0].rm_eo - (matches[0].rm_so+1));
+		printf("%-32s : %s\n", "Boot Page", page);
```
Comment (tswhison) [inline tools/libboard/board_n6010/board_n6010.c:560]: regfree(&re) ?

File: tools/libboard/board_common/board_common.c [hunk 3]
```diff
@@ -500,5 +502,49 @@ fpga_result find_dev_feature(fpga_token token,
...
+
+	retval = fpgaPropertiesGetObjectType(props, &objtype);
+	if (retval != FPGA_OK) {
+		OPAE_ERR("Failed to get object type ");
+		res = retval;
+		goto pro_destroy;
+	}
+
+	if (objtype == FPGA_DEVICE) {
+		printf("%-32s : %s\n", "Boot Page",
```
Comment (tswhison) [inline tools/libboard/board_common/board_common.c:538]: We can only show the boot page for an FPGA_DEVICE?

Comment (anandaravuri) [reply tools/libboard/board_common/board_common.c:538]: OPAE software gets boot page info from bitstream id for  N3000 & D5005 cards , bit 36 is 0 factory or 1 user 

OPAE  software gets boot page info from sysfs  fpga_boot_image for N61010 card.

File: tools/libboard/board_n6010/board_n6010.c (hunk 1 above)
Comment (anandaravuri) [reply tools/libboard/board_n6010/board_n6010.c:555]: fixed

File: tools/libboard/board_n6010/board_n6010.c (hunk 2 above)
Comment (anandaravuri) [reply tools/libboard/board_n6010/board_n6010.c:561]: fixed

**Review by tswhison — APPROVED** [review]

**Pull request merged**